src/main/java/io/cdap/plugin/google/sheets/source/GoogleSheetsSourceConfig.java

```diff
@@ -228,16 +230,47 @@
   @Macro
   private String sheetFieldName;
 
+  public GoogleSheetsSourceConfig(String referenceName, @Nullable String sheetsIdentifiers, String formatting,
+                                  Boolean skipEmptyData, String columnNamesSelection,
+                                  @Nullable Integer customColumnNamesRow, String metadataFieldName,
+                                  Boolean extractMetadata, @Nullable Integer firstFooterRow,
+                                  @Nullable Integer lastHeaderRow, @Nullable Integer lastFooterRow,
+                                  @Nullable String lastDataColumn, @Nullable String lastDataRow,
+                                  @Nullable String metadataCells, @Nullable Integer readBufferSize,
+                                  @Nullable Boolean addNameFields, @Nullable String spreadsheetFieldName,
+                                  @Nullable String sheetFieldName) {
+
+    super(referenceName);
+    this.sheetsIdentifiers = sheetsIdentifiers;
+    this.formatting = formatting;
+    this.skipEmptyData = skipEmptyData;
+    this.metadataFieldName = metadataFieldName;
+    this.columnNamesSelection = columnNamesSelection;
+    this.customColumnNamesRow = customColumnNamesRow;
+    this.extractMetadata = extractMetadata;
+    this.firstFooterRow = firstFooterRow;
+    this.lastHeaderRow = lastHeaderRow;
+    this.lastDataColumn = lastDataColumn;
+    this.lastDataRow = lastDataRow;
+    this.lastFooterRow = lastFooterRow;
+    this.metadataCells = metadataCells;
+    this.readBufferSize = readBufferSize;
+    this.addNameFields = addNameFields;
+    this.spreadsheetFieldName = spreadsheetFieldName;
+    this.sheetFieldName = sheetFieldName;
+  }
+
+
+  public GoogleSheetsSourceConfig(String referenceName) {
+    super(referenceName);
+  }
+
   /**
    * Returns the instance of Schema.
    * @return The instance of Schema
    */
   public Schema getSchema() {
-    if (schema == null) {
-      if (dataSchemaInfo.isEmpty()) {
-        throw new RuntimeException("There are no headers to process. " +
-          "Perhaps no validation step was executed before schema generation.");
-      }
+    if (schema == null && !containsMacro("serviceAccountType")) {
```

Comment:
What happens if service account type is not macro (set to default, i.e. file path), but the file path is a macro?

Reply:
Thanks for catching this. In this scenario the validation would fail. In the next update, I will also check if file path or service acccount json contains Macro.